libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -675,18 +677,31 @@ void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _co
 			continue;
 
 		set<FunctionDefinition const*> ambiguousFunctions;
+		set<FunctionDefinition const*> ambiguousSuperFunctions;
+		set<FunctionDefinition const*> implementedBaseFunctions;
 		SecondarySourceLocation ssl;
 
 		for (;begin != end; begin++)
 		{
 			ambiguousFunctions.insert(*begin);
+			if ((*begin)->isImplemented())
+				implementedBaseFunctions.insert(*begin);
+			FunctionDefinition const* superFunction = (*begin);
+			while (superFunction->annotation().superFunction)
+				superFunction = superFunction->annotation().superFunction;
```

Comment:
We should clarify `superFunction`. The problem is that this is only the super function in the context of the contract that defines the function. If we have another parallel inheritance path, then this could be problematic.

Why does the first overridden function suffice anyway?

Reply:
> Why does the first overridden function suffice anyway?

It doesn't - added a test case that fails for now.